### PR TITLE
added category to data / grid

### PR DESCRIPTION
**Pull request by nathandunn**

### CODE REVIEW

**Comment by nathandunn** [comment]
Fixes #536.    FYI @erasche  

screen shot 2017-03-14 at 4 53 58 pm

I think it as an improvement over what is there, but you can let me know whether or not you find more usable.  I don't have multiple levels of embedding on this, but we didn't have them at all before.  

Anyway, let me know what you think.   Also, not sure if @childers  interested as well.

**Comment by erasche** [comment]
Yep, definitely looks like that would be a huge improvement. (No time to actually test today, sorry)

**Comment by nathandunn** [comment]
Sure.  Just let me know whenever you have a chance to test.

File: src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java [hunk 1]
```diff
@@ -84,6 +71,28 @@ public void setPayload(JSONObject payload) {
...
+        return category;
+    }
+
+    public void setCategory(String category) {
+        this.category = category;
+    }
+
+    public String getStandardCategory(){
+        if(category==null || category.trim().length()==0){
+            return "-----------";
```
Comment (deepakunni3) [inline src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java]: Perhaps some reserved keyword such as `Others` or `Uncategorized`.

File: src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java [hunk 2]
```diff
@@ -84,6 +71,28 @@ public void setPayload(JSONObject payload) {
...
+//            return "        ";
+        }
+        else{
+            String categoryString = "";
+            for(String c : category.split("\\/")){
+                categoryString+=c +"/";
```
Comment (deepakunni3) [inline src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java]: Tried doing `"category": "A/B/C"` but I end up with a category called 'A/B/C' on the Track Panel. I assume this is expected?

**Comment by deepakunni3** [comment]
@nathandunn Definitely an improvement to what was there before.

File: src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java (hunk 1 above)
Comment (nathandunn) [inline src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java]: Agreed.  Kind of struggled with this one.  I think I had Uncategorized before.  I think that makes the most sense.

File: src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java (hunk 2 above)
Comment (nathandunn) [inline src/gwt/org/bbop/apollo/gwt/client/dto/TrackInfo.java]: yes.  That is correct for now.   Maybe could be wrapped in a panel later or something.

**Comment by nathandunn** [comment]
@erasche   Added.  It only does a single level (more than it had done before).    I'm sure it could be pretty easily doctored for even more depth, but that will have to be for another day.

**Pull request merged**